nvdaHelper/remote/ia2LiveRegions.cpp

```diff
@@ -216,18 +246,12 @@ void CALLBACK winEventProcHook(HWINEVENTHOOK hookID, DWORD eventID, HWND hwnd, l
 	if(AccessibleObjectFromEvent(hwnd,objectID,childID,&pacc,&varChild)!=S_OK) {
 		return;
 	}
-	//Retreave the object states, and if its invisible or offscreen ignore the event.
-	VARIANT varState;
-	// #7709: Ensure The VARIANT is initialized, as accState may try to clear it before setting it.
-	VariantInit(&varState);
-	pacc->get_accState(varChild,&varState);
-	VariantClear(&varChild);
-	if(varState.vt==VT_I4&&(varState.lVal&STATE_SYSTEM_INVISIBLE)) {
-		VariantClear(&varState);
+	if (!useIAccessibleForLiveRegionUpdate(pacc, varChild)) {

```

Comment:
Sorry I missed this before, but this must be called further down on the IAccessible2 object instead. QueryInterface to IAccessibleHyperlink will only work once queryServiced to IAccessible2.  Probably best to change its signiture to take an IAccessible2* explicitly rather than an IAccessible* as well, even though the former inherits from the latter.

Reply:
I see, this works differently than the regular COM QI identity. Maybe it would be safer to keep calling accState and accRole on IAccessible proper then. From the docs:

_Although the IAccessible2 IDL is coded such that IAccessible2 is a subclass of MSAA's IAccessible, none of MSAA's IAccessible methods are redefined by IAccessible2._